os/fs/smartfs/smartfs_utils.c

```diff
@@ -1375,10 +1349,25 @@ int smartfs_writeentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s new_
 		goto errout;
 	}
 
+	if (newchain) {
+		/* In case we are writing to a new sector, new_entry.doffset holds the parent logical sector number.
+		   The new sector has to be chained to this parent sector after the entry is written.
+		 */
+
+		/* Chain the next sector into this sector sector */
+		nextsector = psector;
+
+		smartfs_setbuffer(&readwrite, offset, offsetof(struct smartfs_chain_header_s, nextsector), sizeof(uint16_t), (uint8_t *)&nextsector);
```

Comment:
I think we'd better send parentsector instead of newchain as a parameter.
```
int smartfs_writeentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s new_entry, const char *filename, uint16_t type, mode_t mode, struct smartfs_entry_s *direntry, uint16_t parentsector)
{
   ...
   bool newchain = false;
   ...
   if (parentsector != newentry->dsector) {
         newchain = true;
   }
   ....
}

Reply:
Previous implementation did not have new_chain as extra parameter, right?
You prefer to keep it that way if I'm right?

Reply:
This function added in last PR, I can't remember whole of history.
It was acceptable, but now we need information of parentsector, that's all.

Reply:
Changes done